### PR TITLE
Move Integration Tests to openrouteservice Module

### LINKED ISSUE #1345: Move API-Tests into spring-boot IT test suite

### LINKED ISSUE #1349: merge API tests to spring boot integration tests

### PR DESCRIPTION
**Pull request by koebi**
### Tasks

- [x] Move over all tests
- [x] Get rid of duplicate classes in integrationtests package
- [x] Strip down pom.xml (remove out-commented code, unncessary plug-in, unused lifecycle hooks)
- [x] Rename profile "surefire" to something more meaningful
- [x] Document new way of integration testing
- [ ] fix flakey testConditionalAccess and testConditionalSpeed
- [x] fix GetStatusTest waiting for Kafka
- [x] delete openrouteservice-api-tests
- [x] Delete MyIntegrationTest
- [x] investigate `cleanup ors.run for failed builds` in maven-antrun-plugin
- [x] delete uncommented unittest config setup method

### Pull Request Checklist

- [x] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [x] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Closes #1349 
Closes #1345 

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/installation/Configuration.html
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines

### CODE REVIEW

**Comment by MichaelsJP** [comment]
@jlink @koebi One thing that just came up. Would it be possible to disable the debug logging of the integration tests in a normal run, maybe through an ENV variable?

The output is just so cluttered and confusing, and usually not needed in a normal run.

When an action fails, there is the option of re-executing an action in a debug session, which sets `ACTIONS_RUNNER_DEBUG` to true. We could let the integration tests listen to that variable and turn debug on/off conditionally. What do you think?

File: openrouteservice/src/test/java/org/heigit/ors/apitests/centrality/CentralityErrorCodes.java
Comment (sfendrich) [inline openrouteservice/src/test/java/org/heigit/ors/apitests/centrality/CentralityErrorCodes.java:1]: This class is a copy of `org.heigit.ors.centrality.CentralityErrorCodes`. Can we remove the former and use the latter instead? Logically, the error codes are part of the API.

File: openrouteservice/src/test/java/org/heigit/ors/apitests/common/InitializeGraphsOnce.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	private static final String GRAPHS_FOLDER_DELETED = "graphs-folder-deleted";
+
+	@Override
+	public void beforeAll(ExtensionContext extensionContext) {
+		ExtensionContext.Store store = rootStore(extensionContext);
+		boolean graphsFolderAlreadyDeleted = store.getOrDefault(GRAPHS_FOLDER_DELETED, Boolean.class, Boolean.FALSE);
+
+		boolean ciPropertySet = System.getProperty("CI") != null && System.getProperty("CI").equalsIgnoreCase("true");
+		boolean deleteGraphsFolder = !graphsFolderAlreadyDeleted && ciPropertySet;
+		if (deleteGraphsFolder ) {
```
Comment (sfendrich) [inline openrouteservice/src/test/java/org/heigit/ors/apitests/common/InitializeGraphsOnce.java]: Please remove the space before `)`.

File: openrouteservice/src/test/java/org/heigit/ors/apitests/common/InitializeGraphsOnce.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			try {
+				Path graphsFolder = Paths.get(GRAPHS_FOLDER);
+				// Any lower level will not be displayed sind ORS log configuration is not in place at this stage
```
Comment (sfendrich) [inline openrouteservice/src/test/java/org/heigit/ors/apitests/common/InitializeGraphsOnce.java]: sind -> since?

**Review by sfendrich — CHANGES_REQUESTED** [review]
LGTM! Just a few minor comments.

**Comment by jlink** [comment]
log config can be done in the ors.config.json right?

File: openrouteservice/src/test/java/org/heigit/ors/apitests/centrality/CentralityErrorCodes.java
Comment (jlink) [reply openrouteservice/src/test/java/org/heigit/ors/apitests/centrality/CentralityErrorCodes.java:1]: fixed

**Review by sfendrich — APPROVED** [review]

**Review by MichaelsJP — COMMENTED** [review]
Looking good.

**Pull request merged**